shell/platform/windows/window_state.h

```diff
@@ -36,8 +38,31 @@ struct FlutterDesktopPluginRegistrar {
 // Wrapper to distinguish the messenger ref from the engine ref given out
 // in the C API.
 struct FlutterDesktopMessenger {
```

Comment:
`class`

Reply:
as noted above

Reply:
Sorry, I missed this before: per line 16/17 above, there is an explicit goal of not putting any actual code here (to avoid the struct spaghetti that this embedding used to be, and that the GLFW embedder still is). So now that the messenger is a distinct object with a distinct lifetime, it should be a new real class in its own file, just like `FlutterWindowsEngine` is.

The struct can be eliminated entirely in favor of this approach: https://github.com/flutter/engine/blob/a248e624eb855a94fc73c144746d38d53b068271/shell/platform/windows/flutter_windows.cc#L32-L42

Reply:
Oops, it looks like I forgot to unresolve this thread in the last review.

Reply:
Done, it's a bit odd, but done.